db/migrate/20151017221714_create_tag_names.rb

```diff
@@ -0,0 +1,9 @@
+class CreateTagNames < ActiveRecord::Migration
+  def change
+    create_table :tag_names do |t|
+      t.string :name
+      t.references :tag, index: true
```

Comment:
we dont need to add a foreign key to both tables

we only need a foreign key in the tags table